app/models/conditions_response/company_info_incomplete_alert.rb

```diff
@@ -0,0 +1,40 @@
+# This emails all current members in a company if the information for
+# the company is 'incomplete.'
+#
+class CompanyInfoIncompleteAlert < CompanyEmailAlert
+
+
+  # If the required information for a company is missing,
+  # AND there are current members for a company,
+  #
+  # then send the alert if today is in the configuration list of days
+  #
+  # Start sending alerts based on the earliest date that a current member in the company paid their memberhip fee
+  #
+  def send_alert_this_day?(timing, config, company)
+
+    if RequirementsForCoInfoNotComplete.requirements_met?({company: company}) &&
+      !company.current_members.empty?
+
+      earliest_date = company.earliest_current_member_fee_paid
+
+      day_to_check = self.class.days_today_is_away_from(earliest_date, timing)
```

Comment:
Why `self.class`?  Why not explicitly reference the class?

Reply:
makes it more flexible.  That way it can easily be changed/subclassed/etc.  Is a general coding practice. No need to be so specific, no need to restrict the code (= no need to put in the class name) if you don't have to for some good reason. 
#SOLID  "O" - open to change

Reply:
(@patmbolger - just a reminder to give some feedback when you can. didn't want this to get lost in all the other stuff)